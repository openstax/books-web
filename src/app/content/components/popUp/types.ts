import { LinkedArchiveTree, LinkedArchiveTreeNode, LinkedArchiveTreeSection } from '../../types';

export type LocationFilters = Map<
  string,
  { section: LinkedArchiveTreeNode, children?: LinkedArchiveTreeSection[] }
>;

export type LocationFiltersWithChildren = Map<
  string,
  { section: LinkedArchiveTree, children: LinkedArchiveTreeSection[] }
>;

export interface FiltersChange<T> {
  remove: T[];
  new: T[];
}
