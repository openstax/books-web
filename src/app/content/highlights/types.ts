import { Highlight, HighlightsSummary } from '@openstax/highlighter/dist/api';

export type HighlightData = Highlight;

export type CountsPerSource = Exclude<HighlightsSummary['countsPerSource'], undefined>;

export interface State {
  myHighlightsOpen: boolean;
  enabled: boolean;
  focused?: string;
  highlights: null | HighlightData[];
  summary: {
    filters: {
      colors: string[];
      chapters: string[];
    },
    pagination: {
      // even though we're manually splitting our requests out into smaller batches of sources,
      // if a source has more than [numPage] highlights the api will still paginate. in order to
      // access the subsequent pages, we need to store the original sources here, and continue to
      // use them until we run out of pages, then grab the next batch of sources. on the LAST
      // page the response size may be smaller than expected, in this case the next batch of
      // sources should be immediately queried to fill the remaining [numPage]. use
      // paginationUtils.getNextPageSources to get the next batch of sources
      sources: string[];
      page: number;
    } | null,
    loading: boolean;
    // totalCounts should reflect the UNFILTERED state of user's data, used in representing
    // available filter options. this should be updated on create/delete and after auth change
    totalCounts: {[key: string]: number};
    // filteredTotalCounts should reflect the FILTERED state of user's data, used in calculating
    // pagination requests. this should be updated after filter change before requesting
    // filtered highlights (because you need this to query them)
    filteredTotalCounts: {[key: string]: number};
    highlights: {
      [chapterId: string]: {[pageId: string]: HighlightData[]}
    };
  };
}
