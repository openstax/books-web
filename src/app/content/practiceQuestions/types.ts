import { LinkedArchiveTree, LinkedArchiveTreeSection } from '../types';

export interface State {
  isEnabled: boolean;
  summary: PracticeQuestionsSummary | null;
  open: boolean;
  selectedSection: LinkedArchiveTreeSection | null;
  currentQuestionIndex: number | null;
  questions: PracticeQuestion[];
}

export interface PracticeQuestionsSummary {
  countsPerSource: {
    [key: string]: number;
  };
}

interface PracticeAnswer {
  id: number;
  content_html: string;
  correctness: '0.0' | '1.0';
  feedback_html: string;
}

export interface PracticeQuestion {
  group_uuid: string;
  uid: string;
  stem_html: string;
  answers: PracticeAnswer[];
}

export type PracticeQuestions = PracticeQuestion[];

export type PracticeQuestionsLocationFilters = Map<
  string,
  { section: LinkedArchiveTree, children: LinkedArchiveTreeSection[] }
>;
