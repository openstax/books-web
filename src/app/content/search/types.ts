import { SearchResult } from '@openstax/open-search-client';

export interface State {
  results: SearchResult | null;
  loading: boolean;
  query: null | string;
}
