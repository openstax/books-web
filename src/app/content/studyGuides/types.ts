import { HighlightsSummary } from '@openstax/highlighter/dist/api';

export interface State {
  isEnabled: boolean;
  summary: HighlightsSummary | null;
}
