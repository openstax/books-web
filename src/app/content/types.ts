import { RouteParams, RouteState } from '../navigation/types';
import { content } from './routes';
import { State as SearchState } from './search/types';

export interface Params {
  book: string;
  page: string;
}

export interface State {
  tocOpen: boolean | null;
  params?: Params;
  loading: {
    book?: string;
    page?: string;
  };
  search: SearchState;
  book?: Book;
  page?: Page;
  references: PageReferenceMap[];
}

export interface PageReferenceMap extends PageReference {
  match: string;
}

export interface PageReference {
  state: RouteState<typeof content>;
  params: RouteParams<typeof content>;
}

export interface Book {
  id: string;
  shortId: string;
  title: string;
  theme: 'blue' | 'green' | 'gray' | 'yellow' | 'deep-green' | 'light-blue' | 'orange';
  tree: ArchiveTree;
  version: string;
  slug: string;
  license: {
    name: string;
    version: string;
  };
  publish_date: string;
  authors: Array<{
    value: {
      name: string;
    }
  }>;
}

export interface Page {
  id: string;
  shortId: string;
  title: string;
  version: string;
}

export interface ArchiveTreeSection {
  id: string;
  shortId: string;
  title: string;
}

export interface LinkedArchiveTree extends ArchiveTree {
  parent?: LinkedArchiveTree;
}

export interface LinkedArchiveTreeSection extends ArchiveTreeSection {
  parent: LinkedArchiveTree;
}

export interface ArchiveTree extends ArchiveTreeSection {
  contents: Array<ArchiveTree | ArchiveTreeSection>;
}

export interface ArchiveBook {
  id: string;
  shortId: string;
  title: string;
  tree: ArchiveTree;
  version: string;
  license: {
    name: string;
    version: string;
  };
}

export interface ArchivePage {
  id: string;
  shortId: string;
  content: string;
  version: string;
  title: string;
}

export type ArchiveContent = ArchivePage | ArchiveBook;
