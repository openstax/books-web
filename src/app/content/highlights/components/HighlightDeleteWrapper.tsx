import React from 'react';
import { FormattedMessage } from 'react-intl';
import Button from '../../../components/Button';
import * as Styled from './ShowMyHighlightsStyles';

interface HighlightDeleteWrapperProps {
  onCancel: () => void;
  onDelete: () => void;
}

// tslint:disable-next-line:variable-name
const HighlightDeleteWrapper = ({
  onDelete,
  onCancel,
}: HighlightDeleteWrapperProps) => <Styled.HighlightDeleteWrapper>
  <FormattedMessage id='i18n:highlighting:confirmation:delete-both'>
    {(msg: string) => <span>{msg}</span>}
  </FormattedMessage>
  <Styled.HighlightEditButtons>
    <FormattedMessage id='i18n:highlighting:button:delete'>
      {(msg: Element | string) => <Button
        data-testid='delete'
        data-analytics-label='delete'
        size='medium'
        variant='primary'
        onClick={onDelete}
      >{msg}</Button>}
    </FormattedMessage>
    <FormattedMessage id='i18n:highlighting:button:cancel'>
      {(msg: Element | string) => <Button
        size='medium'
        data-analytics-label='cancel'
        data-testid='cancel'
        onClick={onCancel}
      >{msg}</Button>}
    </FormattedMessage>
  </Styled.HighlightEditButtons>
</Styled.HighlightDeleteWrapper>;

export default HighlightDeleteWrapper;
