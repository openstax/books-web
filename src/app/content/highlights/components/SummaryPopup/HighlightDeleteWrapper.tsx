import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import Button from '../../../../components/Button';
import { textStyle } from '../../../../components/Typography';
import { HighlightEditButtons } from './styles';

// tslint:disable-next-line:variable-name
const StyledHighlightDeleteWrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.9);

  span {
    ${textStyle}
    font-size: 16px;
    font-weight: 500;
    line-height: 25px;
    letter-spacing: -0.2;
    color: #fff;
    margin-right: 8px;
  }
`;

interface HighlightDeleteWrapperProps {
  onCancel: () => void;
  onDelete: () => void;
}

// tslint:disable-next-line:variable-name
const HighlightDeleteWrapper = ({
  onDelete,
  onCancel,
}: HighlightDeleteWrapperProps) => <StyledHighlightDeleteWrapper>
  <FormattedMessage id='i18n:highlighting:confirmation:delete-both'>
    {(msg: string) => <span>{msg}</span>}
  </FormattedMessage>
  <HighlightEditButtons>
    <FormattedMessage id='i18n:highlighting:button:delete'>
      {(msg: Element | string) => <Button
        data-testid='delete'
        data-analytics-label='delete'
        size='medium'
        variant='primary'
        onClick={onDelete}
      >{msg}</Button>}
    </FormattedMessage>
    <FormattedMessage id='i18n:highlighting:button:cancel'>
      {(msg: Element | string) => <Button
        size='medium'
        data-analytics-label='cancel'
        data-testid='cancel'
        onClick={onCancel}
      >{msg}</Button>}
    </FormattedMessage>
  </HighlightEditButtons>
</StyledHighlightDeleteWrapper>;

export default HighlightDeleteWrapper;
