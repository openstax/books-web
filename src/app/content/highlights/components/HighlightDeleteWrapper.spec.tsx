import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestStore from '../../../../test/createTestStore';
import MessageProvider from '../../../MessageProvider';
import { Store } from '../../../types';
import HighlightDeleteWrapper from './HighlightDeleteWrapper';

describe('HighlightDeleteWrapper', () => {
  let store: Store;

  beforeEach(() => {
    store = createTestStore();
  });

  it('match snapshot', () => {
    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        {/* tslint:disable-next-line: no-empty */}
        <HighlightDeleteWrapper onCancel={() => {}} onDelete={() => {}} />
      </MessageProvider>
    </Provider>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('properly fire onCancel and onDelete props', () => {
    let deleteClicked = false;
    let cancelClicked = false;

    const component = renderer.create(<Provider store={store}>
      <MessageProvider>
        <HighlightDeleteWrapper
          onCancel={() => { cancelClicked = true; }}
          onDelete={() => { deleteClicked = true; }}
        />
      </MessageProvider>
    </Provider>);

    renderer.act(() => {
      const deleteButton = component.root.findByProps({ 'data-testid': 'delete' });
      deleteButton.props.onClick();
      const cancelButton = component.root.findByProps({ 'data-testid': 'cancel' });
      cancelButton.props.onClick();
    });

    expect(deleteClicked).toEqual(true);
    expect(cancelClicked).toEqual(true);
  });
});
