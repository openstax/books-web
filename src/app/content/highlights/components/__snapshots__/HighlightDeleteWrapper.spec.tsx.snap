// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightDeleteWrapper match snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  min-width: 12rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #fff;
  background-color: #f36b32;
  font-weight: bold;
  border: none;
}

.c2:hover {
  background-color: #e96128;
}

.c2:active {
  background-color: #df571e;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  min-width: 12rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #5e6062;
  background-color: #fff;
  border: 1px solid #d5d5d5;
  font-weight: regular;
}

.c3:hover {
  background-color: #fafafa;
}

.c3:active {
  background-color: #e5e5e5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 8px;
}

.c1 button:first-child {
  margin-right: 8px;
}

.c0 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(0,0,0,0.9);
}

.c0 span {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 16px;
  font-weight: 500;
  line-height: 25px;
  -webkit-letter-spacing: -0.2;
  -moz-letter-spacing: -0.2;
  -ms-letter-spacing: -0.2;
  letter-spacing: -0.2;
  color: #fff;
  margin-right: 8px;
}

<div
  className="c0"
>
  <span>
    Are you sure you want to delete this note and highlight?
  </span>
  <div
    className="c1"
  >
    <button
      className="c2"
      data-analytics-label="delete"
      data-testid="delete"
      focused={true}
      onClick={[Function]}
    >
      Delete
    </button>
    <button
      className="c3"
      data-analytics-label="cancel"
      data-testid="cancel"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
